### PR TITLE
Use JSKit runtime from SwiftPM resources

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Resolves https://github.com/swiftwasm/carton/issues/155.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
As @j-f1 mentioned, we could use https://nodejs.org/dist/latest-v16.x/docs/api/esm.html#loaders to make Node.js entrypoints work with this setup.

File: Package.swift
Comment (MaxDesiatov) [inline Package.swift:36]: This update is required to include https://github.com/vapor/vapor/pull/2823, which allows us to serve `.mjs` files with correct `Content-Type` header.

**Review by j-f1 — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@kateinoigakukun LMK if you're busy, I can either wait until you have to review or merge as is. Thanks!

**Comment by kateinoigakukun** [comment]
Ah, give me 30min for review.

**Comment by MaxDesiatov** [comment]
No worries, I don't want to distract you from your work, but at the same time your input is greatly appreciated, especially as it's a large-ish PR.

**Review by kateinoigakukun — COMMENTED** [review]
We should report diagnostics when users specify JSKit versions that don't have bundled runtime source.

File: Sources/SwiftToolchain/Toolchain.swift
Comment (MaxDesiatov) [reply Sources/SwiftToolchain/Toolchain.swift]: Had to bring this back with a slight modification to check for the version, as you suggested below.

File: Sources/CartonCLI/Commands/TestRunners/NodeTestRunner.swift
Comment (MaxDesiatov) [reply Sources/CartonCLI/Commands/TestRunners/NodeTestRunner.swift:45]: Great idea, that's implemented now.

Comment (kateinoigakukun) [reply Sources/CartonCLI/Commands/TestRunners/NodeTestRunner.swift:45]: Ah, it seems the runtime module is not loaded since it doesn't have package.json

Comment (MaxDesiatov) [reply Sources/CartonCLI/Commands/TestRunners/NodeTestRunner.swift:45]: Actually, just found this: https://nodejs.org/api/esm.html#no-node_path
> NODE_PATH is not part of resolving import specifiers. Please use symlinks if this behavior is desired.

Comment (MaxDesiatov) [reply Sources/CartonCLI/Commands/TestRunners/NodeTestRunner.swift:56]: Reverted to using symlink for making the module available to Node.js in the latest commit

File: Sources/SwiftToolchain/Toolchain.swift
Comment (MaxDesiatov) [inline Sources/SwiftToolchain/Toolchain.swift:204]: Change applied by the formatter.

File: Sources/CartonCLI/Commands/TestRunners/NodeTestRunner.swift
Comment (kateinoigakukun) [reply Sources/CartonCLI/Commands/TestRunners/NodeTestRunner.swift:56]: Ah, ok. I missed that spec 🙏  Seems reasonable to use symlinks

**Review by kateinoigakukun — APPROVED** [review]
LGTM 🦅

**Pull request merged**
